llvm/include/llvm/Support/JSON.h

```diff
@@ -482,6 +482,13 @@ class Value {
   friend class Object;
 
   template <typename T, typename... U> void create(U &&... V) {
+#if defined(LLVM_ADDRESS_SANITIZER_BUILD)
+    // Unpoisoning to prevent overwriting poisoned object (e.g., annotated short
+    // string). Objects that have had their memory poisoned may cause an ASan
```

Comment:
there is a `destroy` method.
could be a bug that this is not called?
I assume std::string will unpoison internal buffer
but if it didn't call ~string, then string with buffer suppose to leak memory?

Reply:
> there is a `destroy` method.
> could be a bug that this is not called?

`std::basic_string` destructor isn't called. I was looking for a while how I can do it better, but I failed.
Just calling `destroy()` in `create` leads to core dump `free(): invalid pointer`.

> but if it didn't call ~string, then string with buffer suppose to leak memory?
Yes. If that string is a long string, that would lead to leaking memory.
Long term, we definitely should fix that, but it seems to not be easy.

I'm still looking at it and trying to find a way to call a destructor.

PS
Scribbling in ADT may lead to leaking memory as well, but it's only in debug mode.

Reply:
I didn't find a way to do it yet. Since this stops a larger commit that, to the best of my knowledge, has no bugs, I'm all for upstreaming this temporary solution and then solving the real problem later.

Short string annotations may help with detecting many bugs, therefore I want them to be in LLVM18. The sooner we upstream them, the sooner bugs are detected.

Reply:
that's fine, but please add a FIXME

Reply:
LGTM